File: Day1_PokemonStatsApp/PokemonStatsApp/Data Models/PokemonListItem.swift
//
//  PokemonListItem.swift
//  PokemonStatsApp
//
//  Created by Sanket Sonje on 15/01/25.
//

import Foundation

/// Represents a single item in the Pokemon list
struct PokemonListItem: Codable, Identifiable {

    /// Unique identifier for the Pokemon
    var id: String { name }

    /// Name of the Pokemon
    let name: String
    
    /// URL containing detailed information about the Pokemon
    let url: String
}

File: Android_PokemonApp/app/src/main/java/com/example/android_pokemonapp/data/model/PokemonListItem.kt
package com.example.android_pokemonapp.data.model

import com.google.gson.annotations.SerializedName

/**
 * Represents a lightweight Pokemon entry in a list.
 *
 * Contains basic information about a Pokemon used in list views and
 * provides a URL to fetch detailed information when needed.
 *
 * @property id Unique identifier of the Pokemon
 * @property name Display name of the Pokemon
 * @property url API endpoint URL to fetch complete Pokemon details
 * @see Pokemon
 */
data class PokemonListItem(
	@SerializedName("id")
	val id: String,

	@SerializedName("name")
	val name: String,

	@SerializedName("url")
	val url: String
)
